### DISCUSSION TITLE
Drop a downstream or upstream task dependency in Airflow

### DISCUSSION THREAD

**Discussion by gmyrianthous** [apache/airflow#29352] [Q&A] [answered]
Assuming we have the two following Airflow tasks in a DAG, 

    from airflow.operators.dummy import DummyOperator
    
    t1 = DummyOperator(task_id='dummy_1')
    t2 = DummyOperator(task_id='dummy_2')

we can specify dependencies as:

```
# Option A
t1 >> t2

# Option B
t2.set_upstream(t1)

# Option C
t1.set_downstream(t2)
```

----------
My question is whether there is any functionality that lets you remove downstream and/or upstream dependencies once they are defined. 

I have a fairly big DAG where most of the tasks (and their dependencies) are generated dynamically. Once the tasks are created, I would like to re-arrange some of the dependencies and/or introduce some new tasks. 

For example, assuming that the functionality implements the following logic

    from airflow.operators.dummy import DummyOperator
    
    t1 = DummyOperator(task_id='dummy_1')
    t2 = DummyOperator(task_id='dummy_2')
    
    t1 >> t2

I would like to then be able to add a new task, add it in between the two tasks, and then remove the old dependency between `t1` and `t2`. Is this possible? 

```
t3 = DummyOperator(task_id='dummy_3')
t1 >> t3
t3 >> t2 
# Somehow remove t1 >> t2
```

**Comment by hussein-awala** [comment, marked as answer]
I am not sure if this works, but you can try it:
```python
t1.downstream_task_ids.pop("dummy_2")
t2.upstream_task_ids.pop("dummy_1")
```

**Reply by gmyrianthous** [reply]
This does work (or `t1.downstream_task_ids.remove('dummy_2')`). 

By any chance, do you know if this imposes any issues (i.e. when removing tasks from `downstream_task_ids` and `upstream_task_ids` property) ?

**Reply by CWKSC** [reply]
```python
AttributeError: 'PlainXComArg' object has no attribute 'upstream_task_ids'
```

Look like it not working in Airflow 3 now
